plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/hms/TestingIcebergHiveMetastoreCatalogModule.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.catalog.hms;
+
+import com.google.inject.Binder;
+import com.google.inject.Scopes;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
+import io.trino.plugin.hive.metastore.DecoratedHiveMetastoreModule;
+import io.trino.plugin.hive.metastore.HiveMetastore;
+import io.trino.plugin.hive.metastore.HiveMetastoreFactory;
+import io.trino.plugin.hive.metastore.RawHiveMetastoreFactory;
+import io.trino.plugin.hive.metastore.thrift.BridgingHiveMetastore;
+import io.trino.plugin.hive.metastore.thrift.ThriftMetastore;
+import io.trino.plugin.hive.metastore.thrift.ThriftMetastoreFactory;
+import io.trino.plugin.iceberg.catalog.IcebergTableOperationsProvider;
+import io.trino.plugin.iceberg.catalog.TrinoCatalogFactory;
+import io.trino.spi.security.ConnectorIdentity;
+
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class TestingIcebergHiveMetastoreCatalogModule
+        extends AbstractConfigurationAwareModule
+{
+    private final HiveMetastore metastore;
+    private final ThriftMetastore thriftMetastore;
+    private final ThriftMetastoreFactory thriftMetastoreFactory;
+    private final HiveMetastoreFactory hiveMetastoreFactory;
```

Comment:
```
private final HiveMetastore metastore;
private final ThriftMetastoreFactory thriftMetastoreFactory;
```

provide these two parameters to the constructor of the module (build them in the test class), so that we get rid of:

```
public HiveMetastore getHiveMetastore()
```

The guice module should not be a factory class.